ERCS/erc-7556.md

```diff
@@ -0,0 +1,157 @@
+---
+eip: 7556
+title: Embedded Accounts as Smart Modules
+description: TODO
+author: Alexander Müller (@alexmmueller), Gregory Markou (@GregTheGreek)
+discussions-to: https://ethereum-magicians.org/t/erc-7556-embedded-accounts-as-smart-modules/16537
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-11-10
+requires: 6900, 7555
+---
+
+## Abstract
+This proposal extends [ERC-xxxx](./ERC-xxxx) to add a new request method allowing applications to register an embedded wallet as [ERC-6900](./ERC-6900) plugin.
+
+## Motivation
+Building off of [ERC-xxxx](./ERC-xxxx), we define a request that allows an application to register an embedded wallet against a users main smart wallet.
+
+The user experience of using a smart account has been dramitically improved with the inclusion of alternative signing methods (such as passkey), but have opened up a few poorer experiences. When an application needs a user to signoff on an a message, to be included via a paymaster, that user needs to access the original signing key that is the owner of the smart account. In a mobile setting, this could be very cumbersome, and include switching contexts (between windows, screens, or apps) in-order to achieve that. Many developers have opted for centalized solutions to manage all the key operations, or simply adding new credentials (such as embedded wallets) as an owner to the smart wallet.
+
+This poses many security concerns for an end user. If a user interacts with two mobile apps, both levereging embedded wallets, and both wallets are added to the smart wallet as owners, the possibility of funds being stolen is high. A malicious app could have an embedded wallet registered on a user wallet, and simply manually drain the account, since they can manipulate the user into signing a malicious payload, or the app simply has access to the private key.
```

Comment:
Yes, but I would say there is a fundamental difference between sharing a privatkey between wallets, and adding many owners with "sudo" to a smart contract wallet

Reply:
To add a new owner to a SA, dApp (game) will require the user to sign at least one userOp with the original signing key. 
So, dApp should find a way for a user to access the original signing key in any case, right?

Instead of adding an owner with 'sudo', a session key with a limited validity timeframe and limited permissions can be added at this point.